internal/xds/translator/translator.go

```diff
@@ -469,7 +462,35 @@ func findXdsEndpoint(tCtx *types.ResourceVersionTable, name string) *endpointv3.
 	return nil
 }
 
-func addXdsCluster(tCtx *types.ResourceVersionTable, args *xdsClusterArgs) error {
+// findXdsSecret finds a xds secret with the same name, and returns nil if there is no match.
+func findXdsSecret(tCtx *types.ResourceVersionTable, name string) *tlsv3.Secret {
+	if tCtx == nil || tCtx.XdsResources == nil || tCtx.XdsResources[resourcev3.SecretType] == nil {
+		return nil
+	}
+
+	for _, r := range tCtx.XdsResources[resourcev3.SecretType] {
+		secret := r.(*tlsv3.Secret)
+		if secret.Name == name {
+			return secret
+		}
+	}
+
+	return nil
+}
+
+func addXdsSecretIfNotExist(tCtx *types.ResourceVersionTable, secret *tlsv3.Secret) error {
```

Comment:
prefer `addXdsSecret` , if we use `addXdsSecretIfNotExist` then we shouldnt be returning a `ErrXdsSecretExists` error

Reply:
Yes, it does return a `ErrXdsSecretExists` if it's already there.